mesonbuild/cargo/manifest.py

```diff
@@ -340,6 +340,13 @@ class Library(BuildTarget['raw.LibTarget']):
     crate_type: T.List[CRATE_TYPE] = dataclasses.field(default_factory=lambda: ['lib'])
     doc_scrape_examples: bool = True
 
+    def __post_init__(self) -> None:
+        # If proc_macro is True, it takes precedence and sets crate_type to proc-macro
+        if self.proc_macro:
+            self.crate_type = ['proc-macro']
+        # If crate_type contains 'proc-macro', that's also valid
+        # (no need to change anything, crate_type is already set correctly)
```

Comment:
I think this needs rebase because those fields are not deal the same way anymore.